src/main/scripts/upgrade_mysql_5_0.sql

```diff
@@ -201,3 +220,20 @@ CREATE TABLE DATAPUBLICATIONFUNDING (
 ALTER TABLE DATAPUBLICATIONFUNDING ADD CONSTRAINT FK_DATAPUBLICATIONFUNDING_DATAPUBLICATION_ID FOREIGN KEY (DATAPUBLICATION_ID) REFERENCES DATAPUBLICATION (ID);
 ALTER TABLE DATAPUBLICATIONFUNDING ADD CONSTRAINT FK_DATAPUBLICATIONFUNDING_FUNDING_ID FOREIGN KEY (FUNDING_ID) REFERENCES FUNDINGREFERENCE (ID);
 ALTER TABLE DATAPUBLICATIONFUNDING ADD CONSTRAINT UNQ_DATAPUBLICATIONFUNDING_0 UNIQUE (DATAPUBLICATION_ID, FUNDING_ID);
+
+-- Add DataCollectionInvestigation table
+
+CREATE TABLE DATACOLLECTIONINVESTIGATION (
+	ID BIGINT AUTO_INCREMENT NOT NULL, 
+	CREATE_ID VARCHAR(255) NOT NULL, 
+	CREATE_TIME DATETIME NOT NULL, 
+	MOD_ID VARCHAR(255) NOT NULL, 
+	MOD_TIME DATETIME NOT NULL, 
+	DATACOLLECTION_ID BIGINT NOT NULL, 
+	INVESTIGATION_ID BIGINT NOT NULL, 
+	PRIMARY KEY (ID)
+);
+
+ALTER TABLE DATACOLLECTIONINVESTIGATION ADD CONSTRAINT DTCLLCTNINVESTIGATIONNVSTGTNID FOREIGN KEY (INVESTIGATION_ID) REFERENCES INVESTIGATION (ID);
+ALTER TABLE DATACOLLECTIONINVESTIGATION ADD CONSTRAINT DTCLLCTNNVESTIGATIONDTCLLCTNID FOREIGN KEY (DATACOLLECTION_ID) REFERENCES DATACOLLECTION (ID);
+ALTER TABLE DATACOLLECTIONINVESTIGATION ADD CONSTRAINT DATACOLLECTIONINVESTIGATION0 UNIQUE (DATACOLLECTION_ID, INVESTIGATION_ID);
```

Comment:
There is a weird result in the database after upgrading from 4.11.  The definition of the `DATACOLLECTIONINVESTIGATION` table looks like:
```sql
CREATE TABLE `DATACOLLECTIONINVESTIGATION` (
  `ID` bigint(20) NOT NULL AUTO_INCREMENT,
  `CREATE_ID` varchar(255) NOT NULL,
  `CREATE_TIME` datetime NOT NULL,
  `MOD_ID` varchar(255) NOT NULL,
  `MOD_TIME` datetime NOT NULL,
  `DATACOLLECTION_ID` bigint(20) NOT NULL,
  `INVESTIGATION_ID` bigint(20) NOT NULL,
  PRIMARY KEY (`ID`),
  UNIQUE KEY `DATACOLLECTIONINVESTIGATION0` (`DATACOLLECTION_ID`,`INVESTIGATION_ID`),
  UNIQUE KEY `UNQ_DATACOLLECTIONINVESTIGATION_0` (`DATACOLLECTION_ID`,`INVESTIGATION_ID`),
  KEY `FK_DATACOLLECTIONINVESTIGATION_INVESTIGATION_ID` (`INVESTIGATION_ID`),
  CONSTRAINT `DTCLLCTNINVESTIGATIONNVSTGTNID` FOREIGN KEY (`INVESTIGATION_ID`) REFERENCES `INVESTIGATION` (`ID`),
  CONSTRAINT `DTCLLCTNNVESTIGATIONDTCLLCTNID` FOREIGN KEY (`DATACOLLECTION_ID`) REFERENCES `DATACOLLECTION` (`ID`),
  CONSTRAINT `FK_DATACOLLECTIONINVESTIGATION_DATACOLLECTION_ID` FOREIGN KEY (`DATACOLLECTION_ID`) REFERENCES `DATACOLLECTION` (`ID`),
  CONSTRAINT `FK_DATACOLLECTIONINVESTIGATION_INVESTIGATION_ID` FOREIGN KEY (`INVESTIGATION_ID`) REFERENCES `INVESTIGATION` (`ID`)
) ENGINE=InnoDB AUTO_INCREMENT=2 DEFAULT CHARSET=utf8mb4;
```
Note that the unique key on `(DATACOLLECTION_ID,INVESTIGATION_ID)` and the foreign keys on `INVESTIGATION_ID` and on `DATACOLLECTION_ID` are defined twice respectively. I believe this is caused by the fact that you used shortened names in the upgrade script and the installation of `icat.server` later on creates the keys again using the long canonical names.

Reply:
Indeed that is weird, as I'm pretty sure I copied the names from the autogenerated tables. I agree that they should be the longer names and I have corrected them.

Reply:
Contemplating that issue, I wondered if we could maybe significantly simplify the upgrade scripts: we could probably drop all create table and all add constraint statements, relying one the fact that the initialization code in `icat.server` will issue the very same statements anyway later on. That would be less error prone. Maybe we should discuss that idea in the upcoming collaboration meeting next week.

Reply:
I agree. I'm not sure why these upgrade scripts are so verbose and duplicate what is being done upon installation of the component anyway.

If I understand correctly, the only statements required in the upgrade scripts would be:
```
ALTER TABLE DATASET ADD FILECOUNT BIGINT;
ALTER TABLE DATASET ADD FILESIZE BIGINT;
ALTER TABLE INVESTIGATION ADD FILECOUNT BIGINT;
ALTER TABLE INVESTIGATION ADD FILESIZE BIGINT;
```
for mysql and the equivalent for oracle, and the rest of the statements could be removed. 

Is that right?

Doing this would mean having to repeat some of the testing but it feels like it would be less error prone like you say.

Reply:
That would at least be my assumption that it is sufficient to keep only the `ALTER TABLE …` statements. But I agree that we need to try it out in practice.